src/main/java/org/folio/circulation/resources/ItemsLostRequiringActualCostResource.java

```diff
@@ -0,0 +1,174 @@
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+public class ItemsLostRequiringActualCostResource extends Resource {
+
+  public final static Collection<String> FIELD_VALUES = List.of(
+    DECLARED_LOST.getValue(),
+    AGED_TO_LOST.getValue());
+
+  private final String rootPath;
+
+  public ItemsLostRequiringActualCostResource(String rootPath, HttpClient client) {
+    super(client);
+    this.rootPath = rootPath;
+  }
+
+  @Override
+  public void register(Router router) {
+    RouteRegistration routeRegistration = new RouteRegistration(rootPath, router);
+    routeRegistration.getMany(this::getMany);
+  }
+
+  private void getMany(RoutingContext routingContext) {
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+    final Repositories repositories = new Repositories(clients);
+    final ItemLostRequiringActualCostsRepresentation itemRepresentation =
+      new ItemLostRequiringActualCostsRepresentation();
+
+    repositories.getItem().findByQuery(matchAny("status.name", FIELD_VALUES))
+      .thenCompose(r -> r.after(items -> 
+        allOf(items, item -> fetchItemLoanAndFeeFine(
+          new ItemLostRequiringCostsEntry(item), repositories, itemRepresentation)
+        )))
+      .thenApply(this::mapResultToJson)
+      .thenApply(r -> r.map(JsonHttpResponse::ok))
+      .thenAccept(context::writeResultToHttpResponse);
+  }
+
+  private CompletableFuture<Result<JsonObject>> fetchItemLoanAndFeeFine(
+    ItemLostRequiringCostsEntry entry, Repositories repositories,
+    ItemLostRequiringActualCostsRepresentation itemRepresentation) {
+
+    return CompletableFuture.completedFuture(Result.succeeded(entry))
+      .thenCompose(r -> r.after(e -> findOpenLoan(e, repositories)))
+      .thenCompose(r -> r.after(e -> mapEntryToJson(e, itemRepresentation)));
+  }
+
+  private CompletableFuture<Result<ItemLostRequiringCostsEntry>> findOpenLoan(
+    ItemLostRequiringCostsEntry entry, Repositories repositories) {
+
+    return repositories.getLoan().findOpenLoanForItem(entry.getItem())
+      .thenComposeAsync(r -> fetchLoanUser(r, repositories.getUser()))
+      .thenComposeAsync(repositories.getAccount()::findAccountsAndActionsForLoan)
+      .thenComposeAsync(repositories.getServicePoint()::findServicePointsForLoan)
+      .thenComposeAsync(repositories.getUser()::findUserForLoan)
+      .thenComposeAsync(repositories.getLoanPolicy()::findPolicyForLoan)
+      .thenComposeAsync(repositories.getOverdueFinePolicy()::findOverdueFinePolicyForLoan)
+      .thenComposeAsync(repositories.getLostItemPolicy()::findLostItemPolicyForLoan)
+      .thenComposeAsync(repositories.getPatronGroup()::findGroupForLoan)
+      .thenApply(r -> r.map(loan -> entry.withLoan(loan)));
```

Comment:
This is neat, thank you for using methods that make it easier to follow the overall flow